chef/cookbooks/ceph/providers/client.rb

```diff
@@ -67,6 +67,9 @@ def auth_set_key(ceph_conf, admin_keyring, keyname, caps)
     Chef::Log.info("Deleting old key with incorrect caps")
     # delete an old key if it exists and is wrong
     Mixlib::ShellOut.new("ceph -k #{admin_keyring} -c #{ceph_conf} auth del #{keyname}").run_command
+    # merge existing caps into new one
+    caps = caps.merge(get_caps(ceph_conf, admin_keyring, keyname))
+    caps = caps["osd"].split(",").collect { |x| x.strip }.uniq.join(",")
```

Comment:
Lint/UselessAssignment: Useless assignment to variable - `caps`.<br>Style/SymbolProc: Pass `&:strip` as an argument to `collect` instead of a block.